### ISSUE TITLE
circleci: failed tests (13382): TestClientNotReady 

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#4630]
The following test appears to have failed:

[#13382](https://circleci.com/gh/cockroachdb/cockroach/13382):

```
E160225 02:54:54.417013 rpc/client.go:328  dial tcp 127.0.0.1:52667: getsockopt: connection reset by peer
--- PASS: TestUnretryableError (0.01s)
=== RUN   TestClientNotReady
W160225 02:54:54.438742 kv/send.go:200  error reply: kv/send.go:286: rpc to Node.Batch: client not ready after 100ns
W160225 02:54:54.446598 kv/send.go:200  error reply: connection is shut down
--- FAIL: TestClientNotReady (0.03s)
	send_test.go:272: failed to send RPC: too many errors encountered (1 of 1 total): connection is shut down
=== RUN   TestComplexScenarios
W160225 02:54:54.469978 kv/send.go:200  error reply: failed to send RPC: test
W160225 02:54:54.470351 kv/send.go:200  error reply: failed to send RPC: test
W160225 02:54:54.470397 kv/send.go:200  error reply: failed to send RPC: test
W160225 02:54:54.470443 kv/send.go:200  error reply: failed to send RPC: test
W160225 02:54:54.470485 kv/send.go:200  error reply: failed to send RPC: test
W160225 02:54:54.470826 kv/send.go:200  error reply: failed to send RPC: test
W160225 02:54:54.470866 kv/send.go:200  error reply: failed to send RPC: test
W160225 02:54:54.471225 kv/send.go:200  error reply: failed to send RPC: test
--
W160225 02:55:11.359133 kv/dist_sender.go:609  failed to invoke BeginTransaction ["a",/Min), Put ["a",/Min), Put ["b",/Min): txn "kv/dist_sender_server_test.go:585 TestPropagateTxnOnPushError" id=d1101823 key="a" rw=true pri=0.01781741 iso=SERIALIZABLE stat=PENDING epo=1 ts=1456368911.356650302,0 orig=1456368911.356650302,0 max=1456368911.356650302,0 failed to push "kv/dist_sender_server_test.go:559 TestPropagateTxnOnPushError.func1" id=b2d08306 key="b" rw=false pri=0.01781746 iso=SERIALIZABLE stat=PENDING epo=0 ts=1456368911.350930639,0 orig=1456368911.350930639,0 max=1456368911.350930639,0
I160225 02:55:11.359358 client/db.go:488  failed batch: txn "kv/dist_sender_server_test.go:585 TestPropagateTxnOnPushError" id=d1101823 key="a" rw=true pri=0.01781741 iso=SERIALIZABLE stat=PENDING epo=2 ts=1456368911.356650302,0 orig=1456368911.356650302,0 max=1456368911.406650302,0 failed to push "kv/dist_sender_server_test.go:559 TestPropagateTxnOnPushError.func1" id=b2d08306 key="b" rw=false pri=0.01781746 iso=SERIALIZABLE stat=PENDING epo=0 ts=1456368911.350930639,0 orig=1456368911.350930639,0 max=1456368911.350930639,0
I160225 02:55:11.407295 storage/replica_command.go:457  auto-gc'ed b2d08306 (1 intents)
I160225 02:55:11.412947 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestPropagateTxnOnPushError (0.27s)
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	18.171s
=== RUN   TestCombinable
--- PASS: TestCombinable (0.00s)
=== RUN   TestBatchSplit
--- PASS: TestBatchSplit (0.00s)
=== RUN   TestFlagsToStr
--- PASS: TestFlagsToStr (0.00s)
=== RUN   TestBatchRequestGetArg
--- PASS: TestBatchRequestGetArg (0.00s)
=== RUN   TestKeyNext
--- PASS: TestKeyNext (0.00s)

```
Please assign, take a look and update the issue accordingly.

**Comment by petermattis** [comment]
Dup of #3942. 

@tamird I've been holding off looking into this under the assumption that it will go away with the switch to gRPC.

**Comment by tamird** [comment]
No way to know! I'll update this when I've finished the grpc work.

**Comment by tamird** [comment]
This version of the bug might have been fixed by #4693, closing for now and reopening #3942.
